### PR TITLE
Add UInt64 with tryAdd/trySubtract methods to manage overflow and underflow

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: units/src/main/java/net/consensys/cava/units/bigints/UInt64Value.java [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+  /**
+   * Returns a value that is {@code (this + value)}.
+   *
+   * @param value The amount to be added to this value.
+   * @return {@code this + value}
+   * @throws ArithmeticException if the result of the addition overflows
+   */
+  default T tryAdd(T value) {
+    T result = add(value);
+    if (this.compareTo(result) > 0) {
```
Comment (cleishm) [inline units/src/main/java/net/consensys/cava/units/bigints/UInt64Value.java]: This check is insufficient to detect overflows that wrap further than the original value.

File: units/src/main/java/net/consensys/cava/units/bigints/UInt64Value.java [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+  /**
+   * Returns a value that is {@code (this + value)}.
+   *
+   * @param value The amount to be added to this value.
+   * @return {@code this + value}
+   * @throws ArithmeticException if the result of the addition overflows
+   */
+  default T tryAdd(long value) {
+    T result = add(value);
+    if (this.compareTo(result) > 0) {
```
Comment (cleishm) [inline units/src/main/java/net/consensys/cava/units/bigints/UInt64Value.java]: This check is insufficient to detect overflows that wrap further than the original value.

File: units/src/main/java/net/consensys/cava/units/bigints/UInt64.java [hunk 3]
```diff
@@ -0,0 +1,793 @@
...
+  /** The value 10 */
+  public final static UInt64 TEN = valueOf(10);
+
+  private static final int INTS_SIZE = 8 / 4;
+  // The mask is used to obtain the value of an int as if it were unsigned.
+  private static final long LONG_MASK = 0xFFFFFFFFL;
+  private static final BigInteger P_2_64 = BigInteger.valueOf(2).pow(64);
+
+  // The unsigned int components of the value
+  private final int ints[];
```
[the remark this answers is not in the document]
Comment (jrhea) [reply units/src/main/java/net/consensys/cava/units/bigints/UInt64.java]: Long would be preferable

[the remark this answers is not in the document]
Comment (cleishm) [reply units/src/main/java/net/consensys/cava/units/bigints/UInt64.java]: @jrhea Well, lucky for you: #174

[the remark this answers is not in the document]
Comment (jrhea) [reply units/src/main/java/net/consensys/cava/units/bigints/UInt64.java]: my man

**Pull request merged**